lib/snapshot.go

```diff
@@ -471,8 +471,6 @@ func (snap *Snapshot) Run() {
 				operation.blockHeight); err != nil {
 				glog.Errorf("Snapshot.Run: Problem adding snapshot chunk to the db")
 			}
-			// Free up a slot in the operationQueueSemaphore, now that a chunk has been processed.
-			<-snap.operationQueueSemaphore
```

Comment:
I'm not the expert on freeing up the slots here, so I'm not sure about this stuff.

Reply:
The existing approach was prone to deadlocking, which caused integration tests to fail. I'm using the integration tests later in the stack, so I had to address the deadlock.